### PR TITLE
WIP: Document classOrObject in Quotes

### PR DESCRIPTION
**Pull request by ahinchman1**
Add more context to `classOrObject` function:

Screen Shot 2019-11-06 at 10 35 11 AM

There's a lot in here to explain these two parameters - I worried the length of the comments might make the file look clunky, but if this is the look we wish to go for I will continue on!

### CODE REVIEW

File: compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt [hunk 1]
```diff
@@ -13,6 +13,55 @@ import org.jetbrains.kotlin.psi.psiUtil.getValueParameters
 import org.jetbrains.kotlin.psi.psiUtil.modalityModifierType
 import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierType
 
+/**
+ * An extension function of [Meta] and inheriting from [ExtensionPhase], [classOrObject] was designed to to feed in
+ * virtually any kind of [KtClass] predicate, followed by mapping a function that takes the desired [Scope] of our
+ * [KtClass] to change whatever PSI elements desired.
+ *
+ * @param [match] designed to to feed in any kind of [KtClass] predicate returning a [Boolean]
```
Comment (ahinchman1) [inline compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt]: Can't seem to figure out why the param keeps shoving down the description to the next line!

Comment (nomisRev) [reply compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt]: `@param`, `@return` etc should be at the of the KDoc as convention, no?

Not sure if this is related to "shoving down the description", I am not sure I understand what you mean  by that :D

Comment (ahinchman1) [reply compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt]: Ah yes, it's happening definitely in the screenshot, regardless of whether I'm doing `@param match ...` or `@param [match] ....`

It should be part of the KDocs yeah!

**Comment by ahinchman1** [comment]
Sorry, I have to close this one and open up a new one because I accidentally worked over Raul's branch. @nomisRev I will link you the new one momentarily!

**Comment by ahinchman1** [comment]
@nomisRev Here it is! Sorry for the mix-up: https://github.com/arrow-kt/arrow-meta/pull/87

File: compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt (hunk 1 above)
Comment (nomisRev) [reply compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt]: I'm sorry I meant:
Shouldn't be at the end/bottom of the KDoc.
[Example in Arrow Fx Concurrent](https://github.com/arrow-kt/arrow/blob/c3fa9ec8567da5bfbd9ecaac2bc055c908dfac00/modules/fx/arrow-fx/src/main/kotlin/arrow/fx/typeclasses/Concurrent.kt#L138)

**Comment by nomisRev** [comment]
All good no problem :)

**Pull request closed without merging**